reasoner/BUILD

```diff
@@ -46,6 +46,7 @@ native_java_libraries(
         "//common:common",
         "//concurrent:concurrent",
         "//concept:concept",
+        "//graph:graph",
         "//logic:logic",
```

Comment:
Need access for the stats.

Reply:
I had a hard think about this and only see one option for how we could avoid the need for this dependency.

The reason it's important to be deliberate here is that opening up a new top-level dependency (look at `docs/package-structure.png`) is always a big risk - you're allowing access between two packages that were otherwise isolated. If you do this without much regard you kind of end up with everything depending on everything, which makes it much harder later to refactor packages & untangle the code later.

The option I see is that we move the `AnswerCountEstimator` into the `logic` package. This would avoid opening up this major new dependency. It also wouldn't by so bad domain-wise because answer estimation of a `ResolvableConjunction` could be located near where `Resolvables` exist.

Then, the estimator could be a member of the `LogicManager`? What do we think of the proposal...

Reply:
The estimator has very little to do with the rest of the contents of the Logic package.
I would imagine there are layers in the packages where higher layers are built upon the lower layers. Adding a dependency between two packages which are clearly in different layers wouldn't lead to cyclic dependencies.
Also, they weren't directly dependent, but they weren't exactly isolated either, given it (and almost every other package) transitively depends on the graph package.

Between moving the AnswerCountEstimator to logic (where it's out of place, given it's effectively a cost-model for planning) and adding a direct edge to a dependency which is already transitively present, I'd expect the latter to be the lesser sin.